torch/_inductor/lowering.py

```diff
@@ -1556,6 +1556,49 @@ def inner_fn(index):
     )
 
 
+@register_lowering(inductor_prims._bucketize, type_promotion_kind=None)
+def _inductor_bucketize(
+    input: TensorBox,
+    boundaries: TensorBox,
+    *,
+    out_int32: bool = False,
+    right: bool = False,
+):
+    assert len(boundaries.get_size()) == 1
+
+    if input.get_device().type != "cuda" or boundaries.get_device().type != "cuda":
+        return fallback_handler(inductor_prims._bucketize, add_to_fallback_set=False)(
+            input, boundaries, out_int32=out_int32, right=right
+        )
+
+    boundaries_size = boundaries.get_size()[0]
+    boundaries_loader = boundaries.make_loader()
+    device = input.get_device()
+    input_loader = input.make_loader()
+
+    index_dtype = torch.int32 if out_int32 else torch.int64
+    triton_dtype = "tl.int32" if out_int32 else "tl.int64"
```

Comment:
Let's not expose the triton_dytpe in the device-agnostic IR.  This will make it harder to extend to non-triton backends.